### DISCUSSION TITLE
Better type matching for HTML attributes between Svelte and Melt-UI

### DISCUSSION THREAD

**Discussion by oscard0m** [melt-ui/melt-ui#789] [Ideas]
After opening these 3 issues:
- https://github.com/melt-ui/melt-ui/issues/764
- https://github.com/melt-ui/melt-ui/issues/787
- https://github.com/melt-ui/melt-ui/issues/838

And getting fixed the same way:
- https://github.com/melt-ui/melt-ui/pull/766
- https://github.com/melt-ui/melt-ui/pull/788
- https://github.com/melt-ui/melt-ui/pull/839

I'm wondering if we could come up with a better way to relate Svelte's TS types with Melt UI's TS types to prevent this kind of mismatches in the future.

---

Also, I would like to open the discussion to catch this type of TS regression for upcoming versions. I think it would be key before getting 1.0, but I'm not sure if TypeScript is a first-class citizen for Melt UI yet.

**Comment by TGlide** [comment]
I agree it's an important situation. Right now we allow anything to be returned from the builder. A potential solution would be to change it to expect SK types to be returned.
